doc/manual/contract.md

```diff
@@ -0,0 +1,329 @@
+
+		# Return the element at the given index.
+		fun get_at(index: Int): T
+		is
+			expect(index < size)
+			ensure(result == self[index])
+		do
+			return self[index]
+		end
+	end
+~~~
+
+As you can see the contracts are defined using several annotations. `invariant` which designates the class invariants, `expect` which designates the preconditions and `ensure` which designates the postconditions. The conditions of a contract must be expressed using boolean expressions in the same way as assertions. Each contract of this example will be detailed in a section below.
+
+All Contracts conditions can be declared in several ways, all are equivalent:
+~~~
+	fun foo(a: Int, b: Int) is ensure(a > 0, b < 0)
+
+	fun foo is ensure(a > 0 and b < 0)
+
+	fun foo is ensure(a > 0), ensure(b < 0)
+~~~
+
+## Writing contracts
+
+Methods, as well as attribute setters, can be associated with two types of contracts, entry contracts precondition (`expect`) and exit contracts postcondition (`ensure`). Together, they define the rules to be respected between the callee and the caller. If the caller fulfills the `expect` condition on entry then the caller can assume that the callee fulfill the `ensure` condition on exit.
+
+All contracts (`expect`, `ensure`) are evaluated in the same context as the called method, it's possible to refer to attributes, methods and parameters in the condition.
+
+When defining a contract (`expect`, `ensure`) on an attribute like bellow, only the automatically generated write property (setter) will use the contract.
+
+~~~
+	var baz: Int is expect(baz > 0) # In the contract, baz refers to the parameter of the write property (setter).
+~~~
+
+### Define preconditions with `expect`
+
+We define a precondition using the annotation `expect`. In the `Stack` class, the method `pop` defines a precondition, that a call to the method should be made on a non-empty stack.
+
+~~~
+	fun pop is expect(not_empty)
+~~~
+
+### Define postconditions with `ensure`
+
+We define a postcondition using the annotation `ensure`. In the `Stack` class, the method `push` defines postconditions. The first guarantee defines that, the method must add the element received as argument (`object`) at the top of the stack.
+
+~~~
+	fun push(object: T) is ensure(self[size - 1] == object)
+~~~
+
+Within an `ensure` contract you can refer to the return value of the method with the keyword `result`. In the `Stack` class, the `get_at` method guarantees that, the return value is the same as the element stored in the stack at the given index.
+
+~~~
+	fun get_at(index: Int): T is ensure(result == self[index])
+~~~
+
+Note that, the keyword `result` can only be used when the method has a return parameter.
+
+
+In `ensure` we can refer to any arguments and instance variables before the method call with the using of `old` keyword. Again, the `pop` method, we can ensure that the `size` of the stack after the method call will be the same as before minus one `old(size) - 1 == size`.
```

Comment:
You use "we" in some places and "you" in other. We should change them all to "we"